addons/website_livechat/static/src/public_models/public_livechat_global.js

```diff
@@ -31,4 +31,11 @@ registerPatch({
             return this._super();
         },
     },
+    fields: {
+        hasWebsiteLivechatFeature: {
```

Comment:
`start()` will be refactored even further. It's called that way just to match closely the legacy structure of code with `Widget` and their `start()` method.

It's not desirable to keep same naming and same split of functions in the near future. The desire architecture of code is declarative with proper definition of models and fields.

That's why it's totally appropriate that this is a field. The implementation detail with `start()` is just technical detail that we don't officially want to keep in the near future.

Reply:
I can understand how this is supposed to be more declarative, but `website_livechat`-specific logic doesn't have much to do in `im_livechat` code. I suggested you to patch the `start` method directly as what you are doing here is adding behaviour to that method anyway, but if you can think of any other way to do it that doesn't require patching the start method, it's also fine by me. My point was primarily to avoid the definition of concepts that are only used by `website_livechat` in `im_livechat`, but apart from that, I don't really have strong opinions as to how it should be done

Reply:
I get the current weirdness of `hasWebsiteLivechatFeature` being defined in `im_livechat`.
But other alternatives that require as much or less efforts and not much better IMO.

Patching `start()` in `website_livechat` is less LOCs, but the patching does something hard to understand and refactor, especially if you don't understand it and/or it's been months after you worked on these LOCs. Overriding methods significantly increases code maintainability, because method cannot just be moved or split in smaller functions (or better: just [computed] fields).

The current solution here is much better for code maintainability, because it's a properly named field and it's easier to make change on code (notably we don't have to deal with huge difficulties when code cleaning with method overrides everywhere).
Also this whole tasks highlights problems with `im_livechat` and `website_livechat` being different and similar. The proper solution is to eventually use `website_livechat` style also for `im_livechat`, but it's too unsafe / functionally inappropriate to do in 16.0.

Also as I said, `start()` method is not a good method/concept to keep in mid/long term. So making code relying on this method and hard to change is certainly not a good thing to do here.